kythe/typescript/indexer.ts

```diff
@@ -1967,6 +1969,24 @@ class Visitor {
     }
   }
 
+  private maybeUnwrapTypeWithArgs(type: ts.Type): ts.Type {
+    if (type.aliasSymbol && type.aliasTypeArguments) {
+      const argumentExtractors =
+        this.host.options.typeWrappers?.[type.aliasSymbol.name];
+      if (argumentExtractors !== undefined) {
+        for (const extractor of argumentExtractors) {
+          let typeArg = type.aliasTypeArguments[extractor.typeParameterIndex];
+          if (!typeArg) continue;
+          if (extractor.transformType) {
+            typeArg = extractor.transformType(this.typeChecker, typeArg);
+          }
+          return typeArg;
```

Comment:
It looks like you are returning the first unwrapped type. Shouldn't you instead return array and then iterate through the array and process each type?

Like instead of

```
type = this.maybeUnwrapTypeWithArgs(type);
```

you do 

```
for (const type of this.maybeUnwrapTypeWithArgs(type)) {
   ...
}